### ISSUE TITLE
input_field type didn't call `prepare`

### ISSUE THREAD

**Issue by cpunion** [rmosolgo/graphql-ruby#936]
I just wrote a mutation:

```ruby
Mutations::DeleteLibrary = GraphQL::Relay::Mutation.define do
  name "DeleteLibrary"

  return_field :result, !types.Boolean

  input_field :id, !types.ID, prepare: Resolvers::GlobalID.new

  resolve ->(obj, args, ctx {
    library = Library.find(args[:id])
    library.destroy!
    true
  }
end
```

When I call it in graphiql:
```graphql
// mutation:

mutation deleteLibrary($input: DeleteLibraryInput!){
  deleteLibrary(input: $input) {
    result
  }
}

// variables:
{
  "id": "TGlicmFyeS0z"
}
```
Some errors happend in rails:
```
ActiveRecord::RecordNotFound (Couldn't find Library with 'id'=TGlicmFyeS0z):
```

But if I call it without variables:
```graphql
mutation {
  deleteLibrary(input: {id: "TGlicmFyeS0z"}) {
    result
  }
}
```

It works fine.

My Resolvers::GlobalID:

```ruby
module Resolvers
  class GlobalID
    def call(globalId, ctx)
      p '==============='
      type_name, item_id = GraphQL::Schema::UniqueWithinType.decode(globalId)
      item_id
    end
  end
end
```

When I call the mutation with variables, the above log didn't print. And if I call it without variables, the log printed.

**Comment by rmosolgo** [comment]
Oh my! What's your graphql-ruby version?

**Comment by cpunion** [comment]
$bundle | grep graphql
Using graphql 1.6.4

Ah. Upgrade first. ^_^!!

**Comment by cpunion** [comment]
Resolved by upgraded to 1.6.7.

Thanks.

**Comment by rmosolgo** [comment]
Glad to hear it!
